web/models/tasks/task.ex

```diff
@@ -21,6 +27,47 @@ defmodule CoursePlanner.Tasks.Task do
     struct
     |> cast(params, @cast_params)
     |> validate_required(@required_params)
-    |> assoc_constraint(:user)
+    |> validate_datetime()
+    |> validate_number(:max_volunteers, greater_than: 0, less_than: 1_000)
   end
+
+  def drop_volunteer(changeset, field, field_data), do: put_assoc(changeset, field, field_data)
+  def update_volunteer(changeset, field, field_data) do
+    changeset
+    |> put_assoc(field, field_data)
+    |> validate_volunteers_limit()
+  end
+
+  defp validate_volunteers_limit(%{valid?: true} = changeset) do
+    max_number_of_volunteers = changeset |> Changeset.get_field(:max_volunteers)
+    number_of_volunteers =
+      changeset
+      |> Changeset.get_field(:volunteers)
+      |> length
+
+    if number_of_volunteers > max_number_of_volunteers do
+      add_error(changeset, :max_volunteers,
+        "The maximum number of volunteers needed for this task is reached")
+    else
+      changeset
+    end
+  end
+  defp validate_volunteers_limit(changeset), do: changeset
+
+  defp validate_datetime(%{valid?: true} = changeset) do
+    finish_time = Changeset.get_field(changeset, :finish_time)
+    start_time = Changeset.get_field(changeset, :start_time)
+    now = Timex.now()
```

Comment:
that's fine for create, grab, and drop 👍  
but as @rhnonose already mentioned this will also prevent any updates to finished tasks unless the coordinator changes the finish time as well